BackendNew/models/students.js

```diff
@@ -481,6 +481,151 @@ exports.getStudents = sendStudents => {
   });
 };
 
+exports.getStudentById = (studentId, sendStudent) => {
+  var basicStudentSQL = `
+  SELECT
+    S.IndexNumber,
+    S.FullName,
+    S.NameWithInitials,
+    S.PhoneNumber,
+    S.Email,
+    S.Sem1GPA,
+    S.Sem2GPA,
+    S.Sem3GPA,
+    S.Sem4GPA,
+    S.SGPA,
+    S.PreferedArea1,
+    S.PreferedArea2,
+    S.PreferedArea3
+  from
+    student S
+  WHERE
+    s.IndexNumber = ?`;
+  var studentCvAndStartDateSQL = `
+  SELECT
+    H.cv,
+    H.DateOfStart,
+    H.BatchId
+  from
+      student S,
+      Student_has_batch H
+  where
+      s.IndexNumber = H.IndexNumber
+      AND s.IndexNumber = ?`;
+  var preferedCompaniesSQL = `
+  SELECT
+    C.Name,
+    SC.batchId
+  FROM
+    student S,
+    Student_select_company SC,
+    Company C
+  WHERE
+    S.IndexNumber = SC.IndexNumber
+    AND SC.CompanyId = C.CompanyId
+    AND S.IndexNumber=?`;
+  var selectedCompanySQL = `
+  SELECT
+    C.Name,
+    SC.batchId,
+    SC.IsSelected
+  FROM
+    student S,
+    Student_select_company SC,
+    Company C
+  WHERE
+    S.IndexNumber = SC.IndexNumber
+    AND SC.CompanyId = C.CompanyId
+    AND SC.IsSelected = 1
+    AND S.IndexNumber = ?
+  `;
+  var internBatchesSQL = `
+  SELECT
+    H.cv,
+    H.BatchId
+  from
+      student S,
+      Student_has_batch H
+  where
+      s.IndexNumber = H.IndexNumber
+      AND s.IndexNumber = ?
+  `;
+  var batchesMap = new Map();
+  var companiesMap = new Map();
+  sql.query(basicStudentSQL, [studentId], (err, sendBasicStudent) => {
+    if (err) {
+      sendStudent(err, null);
+    } else {
+      // To get intern batches.
+      sql.query(internBatchesSQL, [studentId], (err, batches) => {
+        if (err) {
+          sendStudent(err, null);
+        } else {
+          batches.forEach(value => {
+            batchesMap.set(value.BatchId, []);
+            companiesMap.set(value.BatchId, []);
+          });
+          // To get CV and startDate
+          sql.query(
+            studentCvAndStartDateSQL,
+            [studentId],
+            (err, cvAndStartdate) => {
+              if (err) {
+                sendStudent(err, null);
+              } else {
+                cvAndStartdate.forEach(value => {
+                  batchesMap.get(value.BatchId).push({ cv: value.cv });
+                  batchesMap
+                    .get(value.BatchId)
+                    .push({ dateOfStart: value.DateOfStart });
+                });
+                sql.query(
+                  selectedCompanySQL,
+                  [studentId],
+                  (err, selectedCompanies) => {
+                    if (err) {
+                      sendStudent(err, null);
+                    } else {
+                      selectedCompanies.forEach(value => {
```

Comment:
just for clarification. What happen when there is no selected company?

Reply:
For not selected student with one batch
```
{
    "studentdata": {
        "IndexNumber": "164081J",
        "FullName": "Piyumi Dasnayaka",
        "NameWithInitials": "D A P Dasanayaka",
        "PhoneNumber": 717382928,
        "Email": "piyumimadhubashini@gmail.com",
        "Sem1GPA": 4.01,
        "Sem2GPA": 3.48,
        "Sem3GPA": 4.11,
        "Sem4GPA": 3.87,
        "SGPA": 3.87,
        "PreferedArea1": "QA Engineering",
        "PreferedArea2": "Networking",
        "PreferedArea3": "UI/UX"
    },
    "batchInfo": {
        "batch16": {
            "cv": "cv",
            "stratDate": "2020-01-16T18:30:00.000Z",
            "companies": [
                "CreativeSoftware",
                "Virtusa",
                "Wso2"
            ]
        }
    }
}
```

For selected student with more batches
```
{
    "studentdata": {
        "IndexNumber": "164106T",
        "FullName": "Yashod Perera",
        "NameWithInitials": "U L Y G Perera",
        "PhoneNumber": 716931384,
        "Email": "hasinisenanayaka@gmail.com",
        "Sem1GPA": 3.89,
        "Sem2GPA": 3.78,
        "Sem3GPA": 4.03,
        "Sem4GPA": 3.57,
        "SGPA": 3.72,
        "PreferedArea1": "UI/UX",
        "PreferedArea2": "Networking",
        "PreferedArea3": "QA Engineering"
    },
    "batchInfo": {
        "batch16": {
            "cv": "cv",
            "stratDate": "2020-01-16T18:30:00.000Z",
            "selectedCompany": "Virtusa",
            "companies": [
                "Virtusa",
                "Wso2"
            ]
        },
        "batch17": {
            "cv": "cv",
            "stratDate": "2020-01-16T18:30:00.000Z",
            "selectedCompany": "Wso2",
            "companies": [
                "Virtusa",
                "Wso2"
            ]
        }
    }
}
```